### PR TITLE
CORE-10270: remove TransientSoftWrappingKeyMap

### PR DESCRIPTION
**Pull request by dickon**

### CODE REVIEW

**Comment by kyriathar** [comment]
I believe this needs a sibling `corda-api` PR removing configuration:  https://github.com/corda/corda-api/blob/018f117d6cb72b5857b97ce02988f8d687e2d795/data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json#L287?

**Comment by dickon** [comment]
> I believe this needs a sibling `corda-api` PR removing configuration: https://github.com/corda/corda-api/blob/018f117d6cb72b5857b97ce02988f8d687e2d795/data/config-schema/src/main/resources/net/corda/schema/configuration/cryptoLibrary/1.0/corda.cryptoLibrary.json#L287?

I've gone back to keeping the configuration the same, and created https://r3-cev.atlassian.net/browse/CORE-10389 to track config API cleanup. That way this PR is purely refactoring.

**Comment by dickon** [comment]
I also checked with @mnesbit directly about the concept of this PR, i.e. using a zero sized cache instead of eliminating it.

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt [hunk 1]
```diff
@@ -94,28 +95,23 @@ open class SoftCryptoServiceProviderImpl @Activate constructor(
...
+            val name = config.getStringOrDefault("name", KEY_MAP_CACHING_NAME)
             val wrappingKeyMapCacheConfig = SoftCacheConfig(
                 config.getConfig("cache").getLong("expireAfterAccessMins"),
                 config.getConfig("cache").getLong("maximumSize")
             )
             val softWrappingKeyMapConfig = SoftWrappingKeyMapConfig(
-                config.getString("name"),
                 config.getString("salt"),
                 config.getString("passphrase"),
                 wrappingKeyMapCacheConfig
```
Comment (kyriathar) [inline components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt:106]: `SoftWrappingKeyMapConfig.cache` not used, should we remove it? Also should we be renaming it to `MasterWrappingKeyConfig`?

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt [hunk 2]
```diff
@@ -94,28 +95,23 @@ open class SoftCryptoServiceProviderImpl @Activate constructor(
...
             logger.info("set up soft wrapping key map with master key $masterKey")
-            return when (name) {
-                KEY_MAP_TRANSIENT_NAME -> TransientSoftWrappingKeyMap(store, masterKey)
-                KEY_MAP_CACHING_NAME -> CachingSoftWrappingKeyMap(wrappingKeyMapCacheConfig, store, masterKey)
-                else -> throw IllegalStateException(
-                    "Unknown configuration value '$name}' for " +
-                            "${SoftWrappingKeyMap::class.java.simpleName}, must be " +
-                            "$KEY_MAP_TRANSIENT_NAME or $KEY_MAP_CACHING_NAME."
-                )
+            val finalWrappingKeyMapCacheConfig = when (name) {
```
Comment (kyriathar) [inline components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt:110]: Should we move this further up closer to `SoftCacheConfig`, to use a single `wrappingKeyMapCacheConfig` variable and remove `finalWrappingKeyMapCacheConfig`.

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt [hunk 3]
```diff
@@ -94,28 +95,23 @@ open class SoftCryptoServiceProviderImpl @Activate constructor(
...
+                KEY_MAP_TRANSIENT_NAME -> SoftCacheConfig(0, 0)
+                else -> wrappingKeyMapCacheConfig
```
Comment (kyriathar) [inline components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt:112]: Else can be any name, should we throw if it is not one of the expected?

**Review by kyriathar — COMMENTED** [review]
Some minor comments

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt (hunk 3 above)
Comment (dickon) [reply components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt:112]: That was intentional. The plan is to remove the name field in due course, and we have a JIRA for that in the same epic, so for now I don't care what people put in it, nless they use the TRANSIENT value to disble caching.

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt (hunk 1 above)
Comment (dickon) [reply components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt:106]: We should eliminate these config data classes and use SmartConfig directly, there's already a TODO in the code for that. [CORE-10394](https://r3-cev.atlassian.net/browse/CORE-10394) made to track this.

File: components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt (hunk 2 above)
Comment (dickon) [reply components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/SoftCryptoServiceProviderImpl.kt:110]: Don't think so, it's pretty arbitrary but I regard this as grouped with the constructor call on the next line, whereas the upper part of the method is about building up the config objects.

When we remove the config classes that upper code goes away anyway.

**Review by kyriathar — APPROVED** [review]

**Pull request merged**